src/lib/move.ts

```diff
@@ -20,13 +24,36 @@ export default class MoveLib {
     if (toId === fromLineage[1]?.id) {
       throw Error(`Place "${fromLineage[0]?.name}" already has "${toLineage[1]?.name}" as parent`);
     }
-    
-    const { authInfo } = chtApi.chtSession;
-    const url = `http${authInfo.useHttp ? '' : 's'}://${chtApi.chtSession.username}:password@${authInfo.domain}`;
+
+    const jobName = this.getJobName(fromLineage, toLineage);
+    const jobData = this.getJobData(fromId, toId, chtApi);
+    const jobParam: JobParams = {
+      jobName,
+      jobData,
+      queueName: QUEUE_NAMES.MOVE_CONTACT_QUEUE,
+    };
+    await queueManager.addJob(jobParam);
+    const jobsBoardUrl = `/board/queue/${QUEUE_NAMES.MOVE_CONTACT_QUEUE}`;
+
     return {
-      command: `npx cht --url=${url} move-contacts upload-docs -- --contacts=${fromId} --parent=${toId}`,
-      fromLineage,
       toLineage,
+      fromLineage,
+      jobsBoardUrl
+    };
+  }
+
+  private static getJobName(fromLineage: any, toLineage: any): string {
+    return `move_[${fromLineage[0]?.name}]_from_[${fromLineage[1]?.name}]_to_[${toLineage[1]?.name}]`;
+  }
+
+  private static getJobData(fromId: string, toId: string, chtApi: ChtApi): any {
+    const { ENCRYPTION_KEY } = env;
```

Comment:
should you read this from env in `shared/config.ts` along with the other env variables?
can we call it something more descriptive like `QUEUE_PRIVATE_KEY` to differentiate it from `COOKIE_PRIVATE_KEY` which is quite similar in functionality